lib/open_project.rb

```diff
@@ -46,8 +47,23 @@ def self.logger
   end
 
   def self.httpx
-    HTTPX
-      .plugin(:basic_auth)
-      .plugin(:webdav)
+    # It is essential to reuse HTTPX session if persistent connections are enabled.
+    # Otherwise for every request there will be a new connections.
+    # And old connections will not be closed properly which could lead to EMFILE error.
+    Thread.current[:httpx_session] ||= begin
```

Comment:
Won't we run into issues as we did with the workarounds around DelayedJob?

Reply:
You mean in saas environment?
In this case it should be all right to share a session between multiple tenants...

Reply:
True that! :D